cmd/main.go

```diff
@@ -226,17 +226,28 @@ func (c *nodeLocalController) SetupReconcilers(mgr manager.Manager, opts *nnf.Op
 		return err
 	}
 
-	// The NLC controllers relying on the readiness of EC.
+	semNnfLustreMgtDone := make(chan struct{})
+	if err := (&controllers.NnfLustreMGTReconciler{
+		Client:            mgr.GetClient(),
+		Log:               ctrl.Log.WithName("controllers").WithName("NnfLustreMgt"),
+		Scheme:            mgr.GetScheme(),
+		SemaphoreForStart: semNnfNodeStorageDone,
+		SemaphoreForDone:  semNnfLustreMgtDone,
+		ControllerType:    controllers.ControllerRabbit,
+	}).SetupWithManager(mgr); err != nil {
+		return err
+	}
 
 	if err := (&controllers.NnfClientMountReconciler{
 		Client:            mgr.GetClient(),
 		Log:               ctrl.Log.WithName("controllers").WithName("NnfClientMount"),
 		Scheme:            mgr.GetScheme(),
-		SemaphoreForStart: semNnfNodeStorageDone,
+		SemaphoreForStart: semNnfLustreMgtDone,
 		ClientType:        controllers.ClientRabbit,
 	}).SetupWithManager(mgr); err != nil {
 		return err
 	}
+	// The NLC controllers relying on the readiness of EC.
```

Comment:
I think I'm confused about what these comments mean. I thought everything in between this comment and the comment on line 181 were being started sequentially with the semaphores in the reconciler structs. The LustreMGT controller and the ClientMount controller both don't talk to nnf-ec. Are the semaphores there to coordinate something with nnf-ec? Can the LustreMGT and clientmount controllers start at any time because they don't rely on nnf-ec?

Reply:
NnfClientMount reconciler must wait for EC to be ready, otherwise it will not find storage to mount, even if there are resources telling it to do the mount. That's what the comment is saying.

If NnfLustreMGT can do its stuff without waiting for EC to become ready then it can go where you have it, or after NnfClientMount. If it can do its stuff regardless of the state of EC, then maybe it can be the first reconciler to be started at the top of the list, and it wouldn't need to participate in the semaphore thing and wouldn't need a Start() method.